pkg/imgpkg/image/registry.go

```diff
@@ -85,6 +85,23 @@ func (i Registry) Generic(ref regname.Reference) (regv1.Descriptor, error) {
 	return desc.Descriptor, nil
 }
 
+func (i Registry) Get(ref regname.Reference) (*regremote.Descriptor, error) {
+	return regremote.Get(ref, i.opts...)
+}
+
+func (i Registry) Digest(ref regname.Reference) (regv1.Hash, error) {
+	overriddenRef, err := regname.ParseReference(ref.String(), i.refOpts...)
+	if err != nil {
+		return regv1.Hash{}, err
+	}
+	desc, err := regremote.Head(overriddenRef, i.opts...)
```

Comment:
should we return digest directly from the reference if reference is regv1.Digest?

Reply:
This questions seems to want to answer itself. Which is "yes".

But.... I think.... while it might have been a correct digest at some point of time, that artifact might have been deleted from the registry. In which case we would want to make the remote call to "Head" with a reference (which might be a digest) and see it fail with a `404`

Reply:
@cppforlife there is a question as to whether a separate function is introduced (to check if an image has been removed), or additional comments around the Digest needs to be made.